When a bridge port is in blocking state, it does not learn
mac address.  However, allows fdb replace to go through
assuming user knows what is needed.

diff --git a/net/bridge/br_fdb.c b/net/bridge/br_fdb.c
index 991f7a2..0311630 100644
--- a/net/bridge/br_fdb.c
+++ b/net/bridge/br_fdb.c
@@ -697,8 +697,9 @@ static int fdb_add_entry(struct net_bridge_port *source, const __u8 *addr,
 
 	/* skip add if the port is not up*/
 	if (!(state & NUD_PERMANENT || state & NUD_NOARP) &&
-		!(source->state == BR_STATE_LEARNING
-			|| source->state == BR_STATE_FORWARDING))
+		(!(source->state == BR_STATE_LEARNING
+			|| source->state == BR_STATE_FORWARDING) &&
+		 !(flags & NLM_F_REPLACE)))
 		return 0;
 
 	fdb = fdb_find(head, addr, vid);
